thinc/layers/chain.py

```diff
@@ -84,7 +84,8 @@ def init(model: Model, X: Optional[InT] = None, Y: Optional[OutT] = None) -> Non
             break
     for layer in model.layers[:-1]:
         layer.initialize(X=X)
-        X = layer.predict(X)
+        if X is not None:
+            X = layer.predict(X)
```

Comment:
I think we probably want the for loop under the conditional, rather than the conditional on the inside? It should only be the initial `X` that's allowed to be `None`.

Reply:
I actually tried that first because I thought it was more clean, too. But then you have to copy the for loop because the intermediate layers may not get initialized, so you would run them with `layer.initialize()` in the second `for` loop. But that gets more verbose and boils down to the same thing as giving them `X=None`?

Reply:
Hm, I see. Okay!